data_steward/validation/main.py

```diff
@@ -882,53 +883,68 @@ def _is_string_excluded_file(gcs_file_name):
     :hpo_id: hpo from which to copy
     """
+
+    storage_client = StorageClient()
+    # Attempt to acquire hpo bucket, raises google.cloud.exceptions.NotFound
     try:
-        storage_client = StorageClient()
         hpo_bucket = storage_client.get_hpo_bucket(hpo_id)
-        drc_private_bucket = storage_client.get_drc_bucket()
-        source_bucket = storage_client.get_bucket(hpo_bucket)
-        destination_bucket = storage_client.get_bucket(drc_private_bucket)
-        bucket_items = list_bucket(hpo_bucket)
-
-        ignored_items = 0
-        filtered_bucket_items = []
-        for item in bucket_items:
-            item_root = item['name'].split('/')[0] + '/'
-            if item_root.lower() in common.IGNORE_DIRECTORIES:
-                ignored_items += 1
-            else:
-                filtered_bucket_items.append(item)
-
-        logging.info(f"Ignoring {ignored_items} items in {hpo_bucket}")
-
-        prefix = f'{hpo_id}/{hpo_bucket}/'
+    except NotFound:
+        logging.warning(f"Unable to acquire HPO '{hpo_id}' bucket using GCS")
+        return
+    # Attempt to acquire drc bucket, raises google.cloud.exceptions.NotFound
+    try:
+        drc_bucket = storage_client.get_drc_bucket()
+    except NotFound:
+        logging.warning('Unable to acquire DRC bucket using GCS')
+        return
 
-        for item in filtered_bucket_items:
-            item_name = item['name']
-            source_blob = source_bucket.get_blob(item_name)
-            destination_blob_name = f'{prefix}{item_name}'
-            source_bucket.copy_blob(source_blob, destination_bucket,
-                                    destination_blob_name)
+    # Attempt to get item metadata
+    # Raises exception BucketDoesNotExistError, HttpError
+    try:
+        bucket_items: list = list_bucket(hpo_bucket.name)
     except BucketDoesNotExistError as bucket_error:
-        bucket = bucket_error.bucket
+        bucket_name: str = bucket_error.bucket
         # App engine converts an env var set but left empty to be the string 'None'
-        if bucket and bucket.lower() != 'none':
+        if bucket_name and bucket_name.lower() != 'none':
             logging.warning(
-                f"Bucket '{bucket}' configured for hpo_id '{hpo_id}' does not exist"
+                f"Bucket '{bucket_name}' configured for hpo_id '{hpo_id}' does not exist"
             )
         else:
             logging.info(
-                f"Bucket '{bucket}' configured for hpo_id '{hpo_id}' is empty/unset"
+                f"Bucket '{bucket_name}' configured for hpo_id '{hpo_id}' is empty/unset"
             )
+        return
     except HttpError as http_error:
-        message = (
+        message: str = (
             f"Failed to copy files for hpo_id '{hpo_id}' due to the following "
             f"HTTP error: {http_error.content.decode()}")
         logging.exception(message)
+        return
+
+    # Filter
+    ignored_count: int = 0
```

Comment:
This "added" code is actually some of the deletions above, but the addition of exception handling and the movement of other exceptions caused this code to shift.